### PR TITLE
Tests for mentor login flow

### PR DESCRIPTION
**Pull request by callumacrae**
### Context

https://dfedigital.atlassian.net/browse/CPDEL-322
https://dfedigital.atlassian.net/browse/CPDEL-323
https://dfedigital.atlassian.net/browse/CPDEL-356

This mostly already existed, so this was mostly just improving the tests.

### Testing

yes

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module ApplicationHelper
+  def service_name_for(current_user = nil)
+    return "Mentoring for early career teachers" if current_user&.mentor?
```
Comment (kicferk1) [inline app/helpers/application_helper.rb:5]: Let's name the variable `user` or something like that - `current_user` is also a helper method from devise, available in a lot of places, and I don't want to confuse the two.

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    it "returns correct service name if current user is ECT" do
+      mentor_user = create(:user, :early_career_teacher)
+      expect(helper.service_name_for(mentor_user)).to eq("Support for early career teachers")
+    end
+
+    it "returns correct service name if current user is induction coordinator" do
+      mentor_user = create(:user, :induction_coordinator)
+      expect(helper.service_name_for(mentor_user)).to eq("Support for early career teachers")
+    end
```
Comment (kicferk1) [inline spec/helpers/application_helper_spec.rb:30]: Funnily enough, induction coordinators can be mentors too 😭

**Review by kicferk1 — COMMENTED** [review]
Looks pretty good. Did you track down what was causing the issue with factory bot not creating a useful mentor user?

File: spec/helpers/application_helper_spec.rb (hunk 2 above)
Comment (callumacrae) [reply spec/helpers/application_helper_spec.rb:30]: luckily, this one isn't!!

**Comment by callumacrae** [comment]
> Did you track down what was causing the issue with factory bot not creating a useful mentor user?

no idea lol

**Review by kicferk1 — APPROVED** [review]

**Comment by kicferk1** [comment]
> > Did you track down what was causing the issue with factory bot not creating a useful mentor user?
> 
> no idea lol

Fine, in that case I will choose the answer I like. I blame vim, somehow it must be its fault 😂

**Comment by callumacrae** [comment]
> Fine, in that case I will choose the answer I like. I blame vim, somehow it must be its fault 😂

Seems likely 👀

File: spec/helpers/application_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Specs in this file have access to a helper object that includes
+# the ApplicationHelper. For example:
+#
+# describe ApplicationHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
```
Comment (kicferk1) [inline spec/helpers/application_helper_spec.rb:14]: I just noticed, I think this is just a comment for any `:helper` rspec type, right? Can we cut it?

**Review by kicferk1 — COMMENTED** [review]
Sooo, I know you spent a while implementing it, but do we want to remove the dynamic service name now?

**Comment by callumacrae** [comment]
:lolsob:

**Pull request merged**